volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -10,37 +22,59 @@
     exceptions,
     objects,
 )
+
 from volatility3.framework.objects import utility
 from volatility3.framework.symbols.linux import extensions
 
 vollog = logging.getLogger(__name__)
 
 
-class Modules(interfaces.configuration.VersionableInterface):
-    """Kernel modules related utilities."""
+class ModuleInfo(NamedTuple):
+    """
+    Used to track the name and boundary of a kernel module
+    """
+
+    offset: int
+    name: str
+    start: int
+    end: int
+
 
-    _version = (2, 0, 0)
+class ModuleGathererInterface(
+    interfaces.configuration.VersionableInterface, metaclass=ABCMeta
+):
+    _version = (1, 0, 0)
     _required_framework_version = (2, 0, 0)
 
     framework.require_interface_version(*_required_framework_version)
 
-    class ModuleInfo(NamedTuple):
+    gatherer_return_type = Generator[Union[ModuleInfo, "extensions.module"], None, None]
```

Comment:
So this shouldn't be a union...

Reply:
It can be now with isinstance right? Was this comment from before that?

Reply:
Yeah, it was from when I was trying to get rid of the kernel special casing.  WIth an isinstance check instead, this can stay as it is.